osu.Game.Tournament/Screens/StablePathSelectScreen.cs

```diff
@@ -0,0 +1,203 @@
+                                                Origin = Anchor.Centre,
+                                                Width = 300,
+                                                Text = "Auto detect",
+                                                Action = autoDetect
+                                            },
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                },
+                new BackButton
+                {
+                    Anchor = Anchor.BottomLeft,
+                    Origin = Anchor.BottomLeft,
+                    State = { Value = Visibility.Visible },
+                    Action = () => sceneManager?.SetScreen(typeof(SetupScreen))
+                }
+            });
+        }
+
+        private void changePath(Storage storage)
+        {
+            var target = directorySelector.CurrentDirectory.Value.FullName;
+            Logger.Log($"Changing Stable CE location to {target}");
+
+            if (File.Exists(Path.Combine(target, "ipc.txt")))
+            {
+                stableInfo.StablePath.Value = target;
+
+                try
+                {
+                    using (var stream = storage.GetStream(stable_config, FileAccess.Write, FileMode.Create))
+                    using (var sw = new StreamWriter(stream))
+                    {
+                        sw.Write(JsonConvert.SerializeObject(stableInfo,
+                            new JsonSerializerSettings
+                            {
+                                Formatting = Formatting.Indented,
+                                NullValueHandling = NullValueHandling.Ignore,
+                                DefaultValueHandling = DefaultValueHandling.Ignore,
+                            }));
+                    }
+
+                    var fileBasedIpc = ipc as FileBasedIPC;
+                    fileBasedIpc?.LocateStableStorage();
+                    sceneManager?.SetScreen(typeof(SetupScreen));
+                }
+                catch (Exception e)
+                {
+                    Logger.Log($"Error during migration: {e.Message}", level: LogLevel.Error);
+                }
+            }
+            else
+            {
+                overlay = new DialogOverlay();
+                overlay.Push(new IPCErrorDialog("This is an invalid IPC Directory", "Select a directory that contains an osu! stable cutting edge installation and make sure it has an empty ipc.txt file in it."));
+                AddInternal(overlay);
+                Logger.Log("Folder is not an osu! stable CE directory");
+                // Return an error in the picker that the directory does not contain ipc.txt
+            }
+        }
+
+        private void autoDetect()
+        {
+            stableInfo.StablePath.Value = string.Empty; // This forces findStablePath() to look elsewhere.
```

Comment:
`LocateStableStorage` is the only caller of `findStablePath` - so why not just remove the misleading `findFromJsonConfig`, which would render this unnecessary?